keras_hub/src/models/retinanet/box_matcher.py

```diff
@@ -0,0 +1,266 @@
+                    )
+                    matched_values = self._set_values_using_indicator(
+                        matched_values, mask, ind
+                    )
+
+                if self.force_match_for_each_col:
+                    # [batch_size, num_cols], for each column (groundtruth_box),
+                    # find the best matching row (anchor).
+                    matching_rows = ops.argmax(
+                        padded_similarity_matrix,
+                        axis=1,
+                    )
+                    # [batch_size, num_cols, num_rows], a transposed 0-1 mapping
+                    # matrix M, where M[j, i] = 1 means column j is matched to
+                    # row i.
+                    column_to_row_match_mapping = ops.one_hot(
+                        matching_rows, num_rows
+                    )
+                    # [batch_size, num_rows], for each row (anchor), find the
+                    # matched column (groundtruth_box).
+                    force_matched_columns = ops.argmax(
+                        column_to_row_match_mapping,
+                        axis=1,
+                    )
+                    # [batch_size, num_rows]
+                    force_matched_column_mask = ops.cast(
+                        ops.max(column_to_row_match_mapping, axis=1),
+                        "bool",
+                    )
+                    # [batch_size, num_rows]
+                    matched_columns = ops.where(
+                        force_matched_column_mask,
+                        force_matched_columns,
+                        matched_columns,
+                    )
+                    matched_values = ops.where(
+                        force_matched_column_mask,
+                        self.match_values[-1]
+                        * ops.ones([batch_size, num_rows], dtype="int32"),
+                        matched_values,
+                    )
+
+                return ops.cast(matched_columns, "int32"), matched_values
+
+        num_boxes = (
+            similarity_matrix.shape[-1] or ops.shape(similarity_matrix)[-1]
+        )
+        matched_columns, matched_values = ops.cond(
+            pred=ops.greater(num_boxes, 0),
+            true_fn=_match_when_cols_are_non_empty,
+            false_fn=_match_when_cols_are_empty,
+        )
+
+        if squeeze_result:
+            matched_columns = ops.squeeze(matched_columns, axis=0)
+            matched_values = ops.squeeze(matched_values, axis=0)
+
+        return matched_columns, matched_values
+
+    def _set_values_using_indicator(self, x, indicator, val):
+        """Set the indicated fields of x to val.
+
+        Args:
+            x: tensor.
+            indicator: boolean with same shape as x.
+            val: scalar with value to set.
+        Returns:
+            modified tensor.
+        """
+        indicator = ops.cast(indicator, x.dtype)
+        return ops.add(ops.multiply(x, 1 - indicator), val * indicator)
```

Comment:
isn't this just a where op? why are we doing it weird like this

Reply:
Yes, it can be just a where ops. Done changed it.